2-Browsers/Week1/homework/ex1-bookList/index.js

```diff
@@ -18,8 +18,88 @@
 */
 //cspell: enable
 
+class DomElement {
+  constructor(tagType, tagId, cssClass, content) {
+    this.tagType = tagType;
+    this.tagId = tagId !== undefined ? tagId : '';
+    this.cssClass = cssClass !== undefined ? cssClass : '';
+    this.content = content !== undefined ? content : '';
+    this.child = [];
+  }
+  addContent(content) {
+    this.content = content;
+  }
+  addChild(child) {
+    if (Array.isArray(child)) {
+      child.forEach((element) => {
+        this.child.push(element);
+      });
+    } else {
+      this.child.push(child);
+    }
+  }
+  addCssClass(cssClass) {
+    this.cssClass = cssClass;
+  }
+  createDom() {
+    this.domObject = document.createElement(this.tagType);
+    if (this.cssClass !== '') {
+      this.domObject.classList.add(this.cssClass);
+    }
+    if (this.tagId !== '') {
+      this.domObject.id = this.tagId;
+    }
+    this.domObject.innerHTML = this.content;
+    if (this.child.length > 0) {
+      this.child.forEach((element) => {
+        console.log(element);
+        this.domObject.appendChild(element);
+      });
+    }
+    return this.domObject;
+  }
+}
+
+class ImageElement extends DomElement {
+  constructor(tagType, src, alt, tagId, cssClass) {
+    super(tagType, tagId, cssClass);
+    this.src = src;
+    this.alt = alt;
+  }
+  createImageDom() {
+    this.domObject = document.createElement(this.tagType);
+    this.domObject.setAttribute('src', this.src);
+    this.domObject.setAttribute('alt', this.alt);
+    return this.domObject;
+  }
+}
+
 function createBookList(books) {
-  // your code goes in here, return the ul element
+  const ulElement = new DomElement('ul');
+  books.forEach((element) => {
+    const liElement = new DomElement('li');
+    if (element.alreadyRead) {
```

Comment:
Just as an extra challenge: can you refactor this if-statement to a Conditional (ternary) operator?
The goal is to have `liElement.addCssClass` to appear only once.

Reply:
I can delete 
`#bookList ul li.unread {
  background-color: red;
}`
from CSS and consider by default all `#bookList ul li` with background red.
Then in the IF statement just check for (element.alreadyRead) and delete else.